src/snaphu_util.c

```diff
@@ -413,18 +415,18 @@ void **Get2DRowColMem(long nrow, long ncol, int psize, size_t size){
   long row;
   void **array;
 
-  if((array=malloc((2*nrow-1)*psize))==NULL){
+  if((array=(void**)malloc((2*nrow)*psize))==NULL){
     fprintf(sp0,"Out of memory\n");
     exit(ABNORMAL_EXIT);
   }
-  for(row=0; row<nrow-1; row++){
+  for(row=0; row<nrow; row++){
```

Comment:
Hmm...

Reply:
Good question if that's right.

I was "only" merging in updates from others (assuming that their changes are right), not changing myself. I wonder where this comes from...

Reply:
The comment says:

```c
/* function: Get2DRowColMem()
 * --------------------------
 * Allocates memory for 2D array.  The array will have 2*nrow-1 rows.
 * The first nrow-1 rows will have ncol columns, and the rest will
 * have ncol-1 columns.
 */
```

So it's not really a 2D array, but a pair of them stuck together. I think it now overallocates a little, but might be harmless.

Reply:
In the more recent upstream version 2.0.0 it looks different:

https://salsa.debian.org/debian-gis-team/snaphu/-/blob/master/src/snaphu_util.c#L507

(maybe we should trash my PR and switch to that?)

Reply:
See also https://web.stanford.edu/group/radar/softwareandlinks/sw/snaphu/ and https://web.stanford.edu/group/radar/softwareandlinks/sw/snaphu/README_releasenotes.txt. 

There's some hefty changes in 2.0, so it might be worth switching to it.